### PR TITLE
Improve atomix startup and snapshot replication logs

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

* Add more logs at info level during raft startup
* Add more logs for snapshot replication at info level
* Improve error log when appender fails

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: logstreams/src/main/java/io/zeebe/logstreams/impl/log/Listener.java [hunk 1]
```diff
@@ -27,13 +28,17 @@ public void onWrite(final long address) {
...
   public void onWriteError(final Throwable error) {
-    LogStorageAppender.LOG.error(
-        "Failed to append block with last event position {}.", highestPosition, error);
-    if (error instanceof NoSuchElementException) {
+    if (error instanceof NoSuchElementException || error instanceof NoLeader) {
       // Not a failure. It is probably during transition to follower.
+      LogStorageAppender.LOG.warn(
+          "Failed to append block with last event position {}. This can happen during a leader change.",
+          highestPosition,
+          error);
```
Comment (saig0) [inline logstreams/src/main/java/io/zeebe/logstreams/impl/log/Listener.java:36]: Since this case is not a failure, do we want to reduce the log level to `DEBUG` instead?
Or, is it useful to see these log statements on `WARN`?

File: snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotMetadata.java [hunk 2]
```diff
@@ -122,34 +122,37 @@ public String getSnapshotIdAsString() {
 
   @Override
   public int hashCode() {
-    return Objects.hash(getIndex(), getTerm(), getTimestamp());
+    return Objects.hash(index, term, processedPosition, exporterPosition);
```
Comment (saig0) [inline snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotMetadata.java:125]: You changed the `hashCode()` and `equals()` implementation. 
Why did you change it?
Does this change any behavior?

**Review by saig0 — COMMENTED** [review]
Looks good :+1: 

Just, two questions :)

File: snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotMetadata.java (hunk 2 above)
Comment (deepthidevaki) [reply snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedSnapshotMetadata.java:125]: This should have been updated when new fields were added to the class, but it was not.  It won't change any behavior, but it is expected to include processedPosition and exporterPosition when comaparing two snapshots. We did not find out this error, because we are using `SnapshotId::compareTo` to compare them.

File: logstreams/src/main/java/io/zeebe/logstreams/impl/log/Listener.java (hunk 1 above)
Comment (deepthidevaki) [reply logstreams/src/main/java/io/zeebe/logstreams/impl/log/Listener.java]: Agree. DEBUG would be better. I changed it.

**Review by saig0 — APPROVED** [review]
Thanks :tada:
